h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -172,79 +172,82 @@ private Frame prepareLevelOneFrame(String levelOneKey, Model[] baseModels, Frame
-    if (preds == null) {
-      preds =  model.score(frame, predsKey.toString(), null, false);  // no need for metrics here (leaks in client mode)
-      Scope.untrack(preds.keysList());
-    }
-    if (_model._output._base_model_predictions_keys == null)
-      _model._output._base_model_predictions_keys = new Key[0];
+    protected Frame buildPredictionsForBaseModel(Model model, Frame frame) {
+      Key<Frame> predsKey = buildPredsKey(model, frame);
+      Frame preds = DKV.getGet(predsKey);
+      if (preds == null) {
+        preds =  model.score(frame, predsKey.toString(), null, false);  // no need for metrics here (leaks in client mode)
+        Scope.untrack(preds.keysList());
+      }
+      if (_model._output._base_model_predictions_keys == null)
+        _model._output._base_model_predictions_keys = new Key[0];
 
-    if (!ArrayUtils.contains(_model._output._base_model_predictions_keys, predsKey)){
-      _model._output._base_model_predictions_keys = ArrayUtils.append(_model._output._base_model_predictions_keys, predsKey);
+      if (!ArrayUtils.contains(_model._output._base_model_predictions_keys, predsKey)){
+        _model._output._base_model_predictions_keys = ArrayUtils.append(_model._output._base_model_predictions_keys, predsKey);
+      }
+      //predictions are cleaned up by metalearner if necessary
+      return preds;
     }
-    //predictions are cleaned up by metalearner if necessary
-    return preds;
-  }
 
-  protected abstract StackedEnsembleModel.StackingStrategy strategy();
+    protected abstract StackedEnsembleModel.StackingStrategy strategy();
 
-  /**
-   * @RETURN THE FRAME THAT IS USED TO COMPUTE THE PREDICTIONS FOR THE LEVEL-ONE TRAINING FRAME.
-   */
-  protected abstract Frame getActualTrainingFrame();
+    /**
+     * @RETURN THE FRAME THAT IS USED TO COMPUTE THE PREDICTIONS FOR THE LEVEL-ONE TRAINING FRAME.
+     */
+    protected abstract Frame getActualTrainingFrame();
 
-  protected abstract Frame getPredictionsForBaseModel(Model model, Frame actualsFrame, boolean isTrainingFrame);
+    protected abstract Frame getPredictionsForBaseModel(Model model, Frame actualsFrame, boolean isTrainingFrame);
 
-  private Key<Frame> buildPredsKey(Key model_key, long model_checksum, Key frame_key, long frame_checksum) {
-    return Key.make("preds_" + model_checksum + "_on_" + frame_checksum);
-  }
+    private Key<Frame> buildPredsKey(Key model_key, long model_checksum, Key frame_key, long frame_checksum) {
+      return Key.make("preds_" + model_checksum + "_on_" + frame_checksum);
+    }
 
-  protected Key<Frame> buildPredsKey(Model model, Frame frame) {
-    return frame == null || model == null ? null : buildPredsKey(model._key, model.checksum(), frame._key, frame.checksum());
-  }
+    protected Key<Frame> buildPredsKey(Model model, Frame frame) {
+      return frame == null || model == null ? null : buildPredsKey(model._key, model.checksum(), frame._key, frame.checksum());
+    }
 
```

Comment:
I recommend to hide whitespaces in the diff as this file has been reindented (indentation was broken).